lib/stdlib/src/gen_statem.erl

```diff
@@ -845,109 +850,117 @@ loop_hibernate(Parent, Debug, S) ->
        {wakeup_from_hibernate,3}}).
 
 %% Entry point for wakeup_from_hibernate/3
-loop_receive(Parent, Debug, S) ->
+loop_receive(Parent, Debug, #{cancel_timers := 0, hibernate_after := HibernateAfterTimeout} = S) when is_integer(HibernateAfterTimeout) ->
```

Comment:
You are right. I can don't `analyze cancel_timers := 0.` I have been thought that process should not been in hibernate state when It waits timers. But It not necessary. So, I rollback `loop_receive` function and add `after AutoHibernateTimeout` for `receive` operator.